src/main/java/com/google/api/codegen/transformer/ruby/GeneratedLibraryViewModel.java

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.ruby;
+
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.tools.framework.model.Interface;
+import com.google.common.base.Preconditions;
+
+/**
+ * ViewModel for a generated library. This is used by Ruby for the moment, but there's nothing
+ * Ruby-specific about it.
+ */
+// TODO(jcanizales): Move it to/merge it with something common to all languages. Reuse it for Ruby
+// MVVM.
+public class GeneratedLibraryViewModel {
+  private final Interface service;
+  private final SurfaceNamer namer;
+  private final ModelTypeTable typeTable;
+
+  public GeneratedLibraryViewModel(
```

Comment:
View objects are designed to be language-independent and the transformers for each language are suppose to be language-dependent. Adding transformer logic into view classes conflicts with this concept and prevents the flexibility of the fields used by multiple languages.

For example, the field `grpcClassName` in the view class `MockServiceImplView` is used to store both server type name for NodeJS but client type name for PHP:
Node: https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/transformer/nodejs/NodeJSGapicSurfaceTestTransformer.java#L98

PHP:https://github.com/googleapis/toolkit/blob/28aeed1615b8a2417e990088f5d24eab457cc7fe/src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTestTransformer.java#L107

If you find many duplicate code during construction time across different languages, the common way is to create a shared transformer, see `TestCaseTransformer`: 
https://github.com/googleapis/toolkit/blob/master/src/main/java/com/google/api/codegen/transformer/TestCaseTransformer.java